### PR TITLE
Adds upgrade procedures for server

### PR DESCRIPTION
**Pull request by rachfop**
## What does this PR do?

https://github.com/temporalio/documentation/issues/2326

Adds clarification around this.

### CODE REVIEW

File: docs-src/clusters/how-to-upgrade-the-temporal-server-version.md [hunk 1]
```diff
@@ -20,8 +20,25 @@ If a database schema upgrade is required, it will be called out directly in the
 Some releases require changes to the schema, and some do not.
 We ensure that any consecutive versions are compatible in terms of database schema upgrades, features, and system behavior; however there is no guarantee that there is compatibility between _any_ two non-consecutive versions.
 
+### Key considerations
+
+When upgrading the Temporal Server, there are two key considerations to keep in mind:
+
+1. **Sequential Upgrades**: Temporal Server should be upgraded sequentially. That is, if you're on version \(v1.n.x\), your next upgrade should be to \(v1.n+1.x\) or the closest available subsequent version. This sequence should be repeated until your desired version is reached.
```
Comment (flossypurse) [inline docs-src/clusters/how-to-upgrade-the-temporal-server-version.md:27]:
```suggestion
1. **Sequential Upgrades**: Temporal Server should be upgraded sequentially.
That is, if you're on version \(v1.n.x\), your next upgrade should be to \(v1.n+1.x\) or the closest available subsequent version.
This sequence should be repeated until your desired version is reached.
```

Comment (flossypurse) [reply docs-src/clusters/how-to-upgrade-the-temporal-server-version.md:27]: Newline for each sentence.

File: docs-src/clusters/how-to-upgrade-the-temporal-server-version.md [hunk 2]
```diff
@@ -20,8 +20,25 @@ If a database schema upgrade is required, it will be called out directly in the
...
+
+2. **Data Compatibility**: During an upgrade, the Temporal Server either updates or restructures the existing version data to match the data format of the newer version. Temporal Server ensures backward compatibility only between two successive minor versions. Consequently, skipping versions during an upgrade may lead to older data formats becoming unreadable. If the previous data format cannot be interpreted and converted to the newer format, the upgrade process will be unsuccessful.
```
Comment (flossypurse) [inline docs-src/clusters/how-to-upgrade-the-temporal-server-version.md:29]:
```suggestion
2. **Data Compatibility**: During an upgrade, the Temporal Server either updates or restructures the existing version data to match the data format of the newer version.
Temporal Server ensures backward compatibility only between two successive minor versions.
Consequently, skipping versions during an upgrade may lead to older data formats becoming unreadable.
If the previous data format cannot be interpreted and converted to the newer format, the upgrade process will be unsuccessful.
```

Just putting each sentence on new line here as well.

File: docs-src/clusters/how-to-upgrade-the-temporal-server-version.md [hunk 3]
```diff
@@ -20,8 +20,25 @@ If a database schema upgrade is required, it will be called out directly in the
...
+
+### Step-by-Step Upgrade Procedure:
+
+Upgrading the Temporal Server requires a methodical approach to ensure data integrity, compatibility, and seamless transition between versions. The following documentation outlines the step-by-step process to successfully upgrade your Temporal Server.
```
Comment (flossypurse) [inline docs-src/clusters/how-to-upgrade-the-temporal-server-version.md:33]:
```suggestion
Upgrading the Temporal Server requires a methodical approach to ensure data integrity, compatibility, and seamless transition between versions.
The following documentation outlines the step-by-step process to successfully upgrade your Temporal Server.
```

File: docs-src/clusters/how-to-upgrade-the-temporal-server-version.md [hunk 4]
```diff
@@ -20,8 +20,25 @@ If a database schema upgrade is required, it will be called out directly in the
...
+
 When upgrading your Temporal Server version, ensure that you upgrade sequentially.
-For example, when upgrading from v1.n.x, always upgrade to v1.n+1.x (or the next available version) and so on until you get to the required version.
+
+1. **Upgrade Database Schema**: Before initiating the Temporal Server upgrade, use one of the recommended upgrade tools to update your database schema. This ensures it is aligned with the version of Temporal Server you aim to upgrade to.
+2. **Upgrade Temporal Server**: Once the database schema is updated, proceed to upgrade the Temporal Server deployment to the next sequential version.
+3. **Iterative Upgrades** (optional): Continue this process (steps 1 and 2) iteratively until you reach the desired Temporal Server version.
+
+By adhering to the above guidelines and following the step-by-step procedure, you can ensure a smooth and successful upgrade of your Temporal Server.
```
Comment (flossypurse) [inline docs-src/clusters/how-to-upgrade-the-temporal-server-version.md:41]:
```suggestion
1. **Upgrade Database Schema**: Before initiating the Temporal Server upgrade, use one of the recommended upgrade tools to update your database schema.
This ensures it is aligned with the version of Temporal Server you aim to upgrade to.
2. **Upgrade Temporal Server**: Once the database schema is updated, proceed to upgrade the Temporal Server deployment to the next sequential version.
3. **Iterative Upgrades** (optional): Continue this process (steps 1 and 2) iteratively until you reach the desired Temporal Server version.

By adhering to the above guidelines and following the step-by-step procedure, you can ensure a smooth and successful upgrade of your Temporal Server.
```

**Review by flossypurse — COMMENTED** [review]
Suggestions are just newlines for each sentence.

High level this is a great addition.

Let's get an approval from someone on the @temporalio/server team if we can.

**Comment by rachfop** [comment]
Awsome. Fixed and ran yarn gen.

**Review by rodrigozhou — APPROVED** [review]

**Review by flossypurse — APPROVED** [review]

**Pull request merged**